jacobbrunson/RePlay
# Patrick/restructure

# What This Adds

This PR adds the ability to edit assigned prescriptions. A user can change the game, exercise or duration on an assigned prescription card.

# Files Changed

The main changes occurs in CustomPrescriptionsCardView.cs, which adds an event handler to the edit button on an assigned prescription card, and added file EditPrescriptionFragment.cs. A new folder named DataClasses was added to hold the EditPrescriptionEventArgs class. As this class was too simple to be an entity, I put it in its own directory so that other simple struct-like classes may also go there.

# Testing

Some things to try:

1) Delete all the prescriptions. Add a new prescription. Then edit all of its fields.

2) Add a large number of prescriptions (e.g. two or three pages). Edit a prescription on the middle of the second or third page of assigned prescriptions to make sure all works as intended.

3) Delete all the prescriptions again, add a new one to make sure that the prescription can be successfully edited.
